test/conditionalLabel.test.ts

```diff
@@ -0,0 +1,64 @@
+import {shallowMount, mount, Wrapper} from "@vue/test-utils";
+import ConditionalLabel from "../src/ConditionalLabel.vue";
+import {DynamicControlGroup, NumberControl, SelectControl} from "../src/types";
+import Vue from "vue";
+
+describe('Conditional label component', function () {
+
+    const fakeFormGroup: DynamicControlGroup = {
+        label: "Test 1",
+        controls: [
+            {
+                name: "id_1",
+                type: "number",
+                required: true
+            } as NumberControl,
+            {
+                name: "id_2",
+                type: "select",
+                required: true
+            } as SelectControl
+        ]
+    };
+
+    it("renders label if not readonly", () => {
+        const rendered = shallowMount(ConditionalLabel, {
+            propsData: {
+                controlGroup: fakeFormGroup,
+                readonly: false
+            }
+        });
+        const label = rendered.find("label");
+        expect(label.classes()).toStrictEqual(["row", "my-2"]);
```

Comment:
Probably sufficient just to check element exists... comparing classes would make the test unnecessarily brittle to any later changes in styling